chainio/blockbeat.go

```diff
@@ -0,0 +1,78 @@
+package chainio
+
+import (
+	"fmt"
+
+	"github.com/btcsuite/btclog/v2"
+	"github.com/lightningnetwork/lnd/build"
+	"github.com/lightningnetwork/lnd/chainntnfs"
+)
+
+// Beat implements the Blockbeat interface. It contains the block epoch and a
+// buffered error chan.
+//
+// TODO(yy): extend this to check for confirmation status - which serves as the
+// single source of truth, to avoid the potential race between receiving blocks
```

Comment:
yeah basically we have this race, suppose block x closes a channel, then,
1. this spending tx in this block is received by the `chainWatcher`, which will notify the `ChannelArbitrator` about this close event.
2. this block is received by the `chainWatcher`, currently it does nothing
3. this block is received by the `ChannelArbitrator`, which advances its state by launching resolvers and waiting for them to be resolved.

However, there's no guarantee that step 1 will happen first, causing `ChannelArbitrator` to miss the close event in block X, which means we need to wait for one more block to handle the close event.

Tho this is now fixed in the final PR, the APIs `RegisterSpendNtfn/RegisterConfirmationsNtfn` are still hard to use because they put the cognitive load on the caller, so a better API is to do this in the blockbeat - when the beat arrives, we extend the beat to have the ability such as it tells us whether a spending tx is found there.